app/src/main/java/com/ahmettekin/WeatherApp/view/ListFragment.java

```diff
@@ -93,9 +96,24 @@ public void onComplete() {
         );
     }
 
+    public ArrayList<WeatherModel.WeatherItem> filter(ArrayList<WeatherModel.WeatherItem> weatherItemArrayList) {
+
```

Comment:
filter fonksiyonu her çağırıldığında filteredWeatherModel içi temizlenmeli sanırım :)
`filteredWeatherModel.clear()`

Reply:
Harun selamlar. Yorumların için teşekkür ederim :)

Önce filteredWeatherModel değişkenini non-static yaptım ve gördüm ki static olmayınca her defasında fragmentler arası geçişlerde yeniden initialize edildiği için boşalıyordu yani listede her defasında 1 adet şehir gözüküyordu.
Ben de static yaptım bu yüzden. Bu sayede sadece 1 kere initialize olacak. 

Temizleme kısmı da onun kontrolünü yaptım; 

if (temp.getName().matches(ListFragmentArgs.fromBundle(requireArguments()).getCityName()) && !filteredWeatherModel.contains(temp))

yani kullanıcı içinde zaten olan bir şehir ismi girerse listeye eklemiyor.
contains() metodu da arka planda equals() metodunu tetiklediği için WeatherItem sınıfının içinde equals() metodunu isme göre karşılaştırma yapacak şekilde override ettim. Yoksa çalışmıyor :D

filter() fonksiyonunu her defasında temizlediğimde her şehir eklediğimde tek bir şehir gösteriyor.

Reply:
Şimdi şöyle bir durum var;
Eklenen şehir isimleri ArrayListte tutulduğu için uygulama kapandığında siliniyor.
Bu yüzden WeatherItem nesnelerini bir databasede tutiyim diyorum.
Sanırım bunun için WeatherItem sınıfına Serializable interfacesini implemente etmem gerekiyor.
Ne dersin?
:)

Reply:
> Şimdi şöyle bir durum var;
> Eklenen şehir isimleri ArrayListte tutulduğu için uygulama kapandığında siliniyor.
> Bu yüzden WeatherItem nesnelerini bir databasede tutiyim diyorum.
> Sanırım bunun için WeatherItem sınıfına Serializable interfacesini implemente etmem gerekiyor.
> Ne dersin?
> :)

Bunu anlamadım, konuşalım :) 👍